joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForTypesCreator.scala

```diff
@@ -60,16 +61,66 @@ trait AstForTypesCreator { this: AstCreator =>
     astExprOfCommand ++ Seq(bodyBlockAst)
   }
 
-  def astForClassOrModuleReferenceContext(
-    ctx: ClassOrModuleReferenceContext,
-    baseClassName: Option[String] = None
-  ): Seq[Ast] = {
-    val className = ctx.className(baseClassName)
+  def astForModuleDefinitionPrimaryContext(ctx: ModuleDefinitionPrimaryContext): Seq[Ast] = {
+    val className = ctx.moduleDefinition().classOrModuleReference().className(None)
 
     if (className != Defines.Any) {
       classStack.push(className)
+
+      val fullName = classStack.reverse.mkString(pathSep)
+      val namespaceBlock = NewNamespaceBlock()
+        .name(className)
+        .fullName(fullName)
+        .filename(filename)
+
+      val moduleBodyAst = astInFakeMethod(className, fullName, filename, ctx)
+
+      if (classStack.size > 0) {
```

Comment:
I would remove this check since if, somewhere along the line we popped where we shouldn't have, we want to fix it.